### ISSUE TITLE
Errors when changing tag hierarchy in main navigation

### ISSUE THREAD

**Issue by mikemurray** [reactioncommerce/reaction#1312]
### Expected behavior

When adding an existing tag that is a tag group, a nested child of a top level navigation item to the main navigation, there should be no errors.
### Actual Behavior

Errors appear and become persistent.

```
Exception in template helper: TypeError: Cannot read property 'relatedTagIds' of undefined
Exception in template helper: TypeError: Cannot read property 'isEditing' of undefined
```
### Steps to Reproduce the Behavior
1. Create a new top-level navigation item
2. Expand drop down
3. Add a `Tag Group` with a new name
4. Add an item to that tag group
5. In the main navigation, add a new top-level, use the same name you used for the `Tag Group`
6. Be amazed as your js console blows up with errors.

**Comment by zenweasel** [comment]
Verified that this error still happens in version 1.1.1.

**Comment by impactmass** [comment]
fix going in #2251

**Comment by impactmass** [comment]
resolved in #2306
